model/client4.go

```diff
@@ -36,10 +37,15 @@ func NewAPIv4Client(url string) *Client4 {
 }
 
 func BuildErrorResponse(r *http.Response, err *AppError) *Response {
-	if r == nil {
-		return &Response{StatusCode: 0, Error: err}
-	} else {
-		return &Response{StatusCode: r.StatusCode, Error: err}
+	statusCode := 0
+	if r != nil {
+		statusCode = r.StatusCode
+	}
+
+	return &Response{
+		StatusCode: statusCode,
+		Error:      err,
+		Response:   r,
```

Comment:
Another note: It's now redundant to provide both the status code and the response in this object. I would say it'd be better off without the status code field, but that's an extremely minor thing and it was there before, so whatever.

I don't have an opinion on @crspeller's comment. Generally when I see an error, it's pretty intuitive to me that the things packages alongside it are mutually exclusive.